src/renderers/dom/client/syntheticEvents/SyntheticEvent.js

```diff
@@ -90,17 +100,21 @@ function SyntheticEvent(dispatchConfig, targetInst, nativeEvent, nativeEventTarg
 assign(SyntheticEvent.prototype, {
 
   preventDefault: function() {
+    if (this._destructored) {
+      if (__DEV__) {
+        warning(
+          event,
+          'This synthetic event is reused for performance reasons. If you\'re ' +
+          'seeing this, you\'re calling `preventDefault` on a ' +
+          'released/nullified synthetic event. This is a no-op. See ' +
+          'https://fb.me/react-event-pooling for more information.'
+        );
+      }
+      return;
+    }
+
     this.defaultPrevented = true;
     var event = this.nativeEvent;
-    if (__DEV__) {
```

Comment:
Can you please help me understand why it is necessary to move this block up and wrap it in `if (this._destructored)`? It seems to me like existence of `nativeEvent` was enough and still should be enough to tell whether the object was “cleared”, which is how the old code worked.